components/telemetry-operator/controller/tracepipeline/resources.go

```diff
@@ -17,6 +17,7 @@ import (
 const (
 	basicAuthHeaderVariable = "BASIC_AUTH_HEADER"
 	otlpEndpointVariable    = "OTLP_ENDPOINT"
+	configHashAnnotationKey = "checksum/config"
```

Comment:
What do you think about a maybe convention driven key for the annotation inspired by the  `kubectl.kubernetes.io/last-applied-configuration`, in our case something like `kyma-project.io/current-config-sha`?